flaml/nlp/autotransformers.py

```diff
@@ -0,0 +1,901 @@
+from functools import partial
+
+from .dataset.metric_auto import get_default_and_alternative_metric
+from .dataset.submission_auto import auto_output_prediction
+from .hpo.hpo_searchspace import AutoHPOSearchSpace, HPO_SEARCH_SPACE_MAPPING
+from .huggingface.modeling_auto import AutoSeqClassificationHead
+from .utils import PathUtils, _variable_override_default_alternative
+from .hpo.grid_searchspace_auto import AutoGridSearchSpace
+from .hpo.searchalgo_auto import AutoSearchAlgorithm, SEARCH_ALGO_MAPPING
+from .hpo.scheduler_auto import SCHEDULER_MAPPING, AutoScheduler
+from .hpo.grid_searchspace_auto import GRID_SEARCH_SPACE_MAPPING
+from dataclasses import dataclass, field
+from typing import Any, Dict, List, Optional, Tuple
+
+from .huggingface.trainer import TrainerForAutoTransformers
+
+import logging
+logger = logging.getLogger(__name__)
+logger_formatter = logging.Formatter(
+    '[%(name)s: %(asctime)s] {%(lineno)d} %(levelname)s - %(message)s',
+    '%m-%d %H:%M:%S')
+
+task_list = [
+    "text-classification",
+    "text-regression",
+    "question-answering"
+]
+
+
+class AutoTransformers:
+
+    '''The AutoTransformers class
+
+    Example:
+
+        .. code-block:: python
+
+            autohf = AutoTransformers()
+            autohf_settings = {"metric_name": "accuracy",
+                   "mode_name": "max",
+                   "resources_per_trial": {"gpu": 4, "cpu": 4},
+                   "wandb_key": wandb_key,
+                   "search_algo_name": method,
+                   "num_samples": 4,
+                   "time_budget": 7200,
+                   "points_to_evaluate": [{
+                       "num_train_epochs": 1,
+                       "per_device_train_batch_size": 128, }]
+                   }
+
+            autohf.fit(train_dataset,
+                       eval_dataset,
+                       **autohf_settings,)
+
+    '''
+
+    _task_name: str = field(default=None, metadata={"help": "task name"})
+    _dataset_name: list = field(default=None, metadata={"help": "dataset name"})
+    _subdataset_name: str = field(default=None, metadata={"help": "dataset name"})
+    _model_type: str = field(default=None, metadata={"help": "huggingface type."})
+    _split_mode: str = field(default=None, metadata={"help": "The submit mode."})
+
+    _scheduler_name: str = field(default=None, metadata={"help": "The scheduler name."})
+    _search_algo_name: str = field(default=None, metadata={"help": "The hpo method."})
+
+    _num_labels: Optional[int] = field(default=None, metadata={"help": "number of labels"})
+    _metric_name: str = field(default=None, metadata={"help": "metric name"})
+    _metric_mode_name: str = field(default=None, metadata={"help": "metric mode name"})
+
+    _max_seq_length: Optional[int] = field(default=None, metadata={"help": "max seq length"})
+
+    _fp16: Optional[bool] = field(default=True, metadata={"help": "is fp16"})
```

Comment:
in general, the help info needs to be expanded a little for many of these options.